### PR TITLE
Update resolver to handle new gdk interface changes

### PR DESCRIPTION
**Pull request by jgriffiths**
Requires the associated gdk changes

### CODE REVIEW

File: green_cli/authenticators/__init__.py [hunk 1]
```diff
@@ -140,6 +140,7 @@ def default_hw_device_info(self):
                   'name': self.name,
                   'supports_low_r': False,
                   'supports_liquid': 0,
+                  'supports_host_unblinding': True,
```
Comment (JamieDriver) [inline green_cli/authenticators/__init__.py:143]: Not sure this is the correct place for this flag.  
May instead want it set in WallyAuthenticatorLiquid (ie wally.py ~ 160)

File: green_cli/authenticators/__init__.py [hunk 2]
```diff
@@ -178,19 +179,16 @@ def resolve(self, details):
             return response
         if details['action'] == 'sign_tx':
             return self.sign_tx(details)
-        if details['action'] == 'get_receive_address':
-            blinding_script_hash = bytes.fromhex(details['address']['blinding_script_hash'])
-            public_blinding_key = self.get_public_blinding_key(blinding_script_hash)
-            return json.dumps({'blinding_key': public_blinding_key.hex()})
+        if details['action'] == 'get_master_blinding_key':
+            return json.dumps({'master_blinding_key': self.self.master_blinding_key.hex()})
```
Comment (jkauffman1) [inline green_cli/authenticators/__init__.py:183]: is this reference to `self.self` right?

File: green_cli/authenticators/__init__.py (hunk 1 above)
Comment (jgriffiths) [reply green_cli/authenticators/__init__.py:143]: Agreed, updated.

File: green_cli/authenticators/__init__.py (hunk 2 above)
Comment (jgriffiths) [reply green_cli/authenticators/__init__.py:183]: No, last minute faffing. Will mark this PR as WIP until development/testing is complete.

**Comment by JamieDriver** [comment]
Testing this locally with wip changes to support Jade hww appears to be working as desired.  
- agree local caching of nonces now working  
- master-key denied falls back to asking Jade for blinding keys, nonces etc.  
- master-key provided - no blinding keys nor nonces are requested from Jade at during get-new-address, create-tx, etc.  
(Note: much deleting of local .sqllite file to ensure *not* benefiting from fixed nonce caching as above!)

**Comment by jkauffman1** [comment]
utack b1c8744a7acc46112d5f0b9a478e23d202f20b18

File: green_cli/authenticators/__init__.py [hunk 3]
```diff
@@ -178,28 +179,18 @@ def resolve(self, details):
...
+                public_keys.append(self.get_public_blinding_key(bytes.fromhex(script)).hex())
+            return json.dumps({'public_keys': public_keys})
+        if details['action'] == 'get_blinding_nonces':
             nonces = []
-            for elem in details['blinded_scripts']:
-                pubkey = bytes.fromhex(elem['pubkey'])
-                script = bytes.fromhex(elem['script'])
-                nonces.append(self.get_shared_nonce(pubkey, script).hex())
-            retval['nonces'] = nonces
+            for (pubkey, script) in zip(details['public_keys'], details['scripts']):
```
Comment (JamieDriver) [inline green_cli/authenticators/__init__.py:192]: Seems to be an issue with the 'required_data' containing 'None' for `public_keys` and `scripts` when a) host-unblinding is not active or key-denied, and b) all nonces already cached (by prior call).  
Suspect it is an issue with the data being passed, so put more details on the related gdk branch MR.

Comment (jgriffiths) [reply green_cli/authenticators/__init__.py:192]: Yes, fixed on the gdk side, thanks!

**Pull request merged**